dist/tools/pyterm/pyterm

```diff
@@ -791,6 +803,9 @@ if __name__ == "__main__":
                         "of CR and LF. Examples: -nl=LF, -nl=CRLF. "
                         "(Default is %s)" % defaultnewline,
                         default=defaultnewline)
+    parser.add_argument("--repeat-command-on-empty-line",
+                        help="Enable repeating command on empty line",
+                        action="store_true", default=False)
```

Comment:
To keep the existing behaviour, the default should be True. IIUC, you can change this via PYTERMFLAGS in your test if you want.

Reply:
It is done on purpose, as I want the behavior to be the same when you do `make test`, make term`, and use any other `RIOT_TERMINAL`.

And to do the opposite, it cannot just be flipped, but the option would need to be renamed to a "do not repeat command on empty line" as it is an option without argument.

Reply:
Indeed, but in this case, there's no need for `default=False` since it's set automatically when using `action="store_true"`.

Anyway, as some others, I wouldn't invert this behaviour in this PR just because it's more convenient for tests. The goal of the tool is to be convenient for users and if users tend to prefer the "repeat command" on empty line, this should be the default. The test can be adapted.

Reply:
> It is done on purpose, as I want the behavior to be the same when you do `make test`, `make term`, and use any other `RIOT_TERMINAL`.
> 
> And to do the opposite, it cannot just be flipped, but the option would need to be renamed to a "do not repeat command on empty line" as it is an option without argument.

Please remember, that the testing experience is not the primary user experience, but the normal terminal is.

Reply:
Yes, then having different behavior depending on the board, example (ethos), or RIOT_TERMINAL is for me bad as user experience.

Reply:
> Indeed, but in this case, there's no need for `default=False` since it's set automatically when using `action="store_true"`.

I learned something.

Reply:
> Yes, then having different behavior depending on the board, example (ethos), or RIOT_TERMINAL is for me bad as user experience.

So does just silently deactivating a feature many people rely on for their local testing.

Reply:
> I learned something.

so did I :)

Reply:
> > Yes, then having different behavior depending on the board, example (ethos), or RIOT_TERMINAL is for me bad as user experience.
> 
> So does just silently deactivating a feature many people rely on for their local testing.

Those are different tools BTW. Especially `ethos` and `native` could also use `pyterm` by wrapping around with TCP, however this would just add unnecessary overhead when doing this in the default case.

Reply:
"unnecessary" agreed.

What happened to me, was more when typing a "ping" command, I would type "enter" to add some spacing, and oh, it would be repeated many times.
And it cannot be interrupted with `ctrl+c`.
I usually do not call it a feature when it can be done explicitly with just "up + enter".

Reply:
> I usually do not call it a feature when it can be done explicitly with just "up + enter".

Again, "up + enter" is very unreliable if you are doing flood testing with commands that are only doing one thing at a time...